plugins/package-managers/maven/src/main/kotlin/utils/MavenParsers.kt

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2017 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.maven.utils
+
+import java.lang.invoke.MethodHandles
+
+import org.apache.logging.log4j.kotlin.loggerOf
+import org.apache.maven.model.Scm
+import org.apache.maven.project.MavenProject
+
+import org.ossreviewtoolkit.downloader.VcsHost
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.utils.parseRepoManifestPath
+import org.ossreviewtoolkit.utils.common.splitOnWhitespace
+import org.ossreviewtoolkit.utils.common.withoutPrefix
+import org.ossreviewtoolkit.utils.ort.DeclaredLicenseProcessor
+import org.ossreviewtoolkit.utils.ort.ProcessedDeclaredLicense
+import org.ossreviewtoolkit.utils.spdx.SpdxOperator
+
+// See http://maven.apache.org/pom.html#SCM.
+private val SCM_REGEX = Regex("scm:(?<type>[^:@]+):(?<url>.+)")
+private val USER_HOST_REGEX = Regex("scm:(?<user>[^:@]+)@(?<host>[^:]+)[:/](?<path>.+)")
+
+private val logger = loggerOf(MethodHandles.lookup().lookupClass())
+
+/**
+ * When asking Maven for the SCM connection or the SCM URL of a POM that does not itself define these values,
+ * Maven returns the values of the first parent POM (if any) that defines one and appends the artifactIds of all
+ * child POMs to it, separated by slashes.
+ * This behavior is fundamentally broken because it invalidates the SCM connection / URL for all VCS that cannot
+ * limit cloning to a specific path within a repository, or use a different syntax for that. Also, the
+ * assumption that the source code for a child artifact is stored in a top-level directory named like the
+ * artifactId inside the parent artifact's repository is often not correct.
+ * To address this, determine the SCM connection and URL of the parent (if any) that is closest to the root POM
+ * and whose SCM connection / URL still is a prefix of the child POM's SCM values.
```

Comment:
commit-msg: Change should be marked breaking.

While at it, would it make sense to reduce to `internal` visibility?

Reply:
See the last commit 😉

Reply:
This commit should be marked breaking anyway.